### PR TITLE
Move image classification example to the docs

### PR DESCRIPTION
**Pull request by MKhalusova**
This PR moves the [fine-tuning image classifier with LoRA](https://github.com/huggingface/peft/blob/main/examples/image_classification/image_classification_peft_lora.ipynb) example from the existing notebook to documentation with some minor text improvements. 
Other examples will follow.

### CODE REVIEW

**Comment by MKhalusova** [comment]
cc @stevhliu

File: docs/source/task_guides/image_classification_lora.mdx [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+```python
+import numpy as np
+import evaluate
+
+metric = evaluate.load("accuracy")
+
+
+# the compute_metrics function takes a Named Tuple as input:
+# predictions, which are the logits of the model as Numpy arrays,
+# and label_ids, which are the ground-truth labels as Numpy arrays.
```
Comment (stevhliu) [inline docs/source/task_guides/image_classification_lora.mdx:301]: I think it'd be easier to read if this was regular text and not included in the code example

File: docs/source/task_guides/image_classification_lora.mdx [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+def compute_metrics(eval_pred):
+    """Computes accuracy on a batch of predictions"""
+    predictions = np.argmax(eval_pred.predictions, axis=1)
+    return metric.compute(predictions=predictions, references=eval_pred.label_ids)
+```
+
+## Define collation function
+
+A collation function is used by `Trainer` to gather a batch of training and evaluation examples and prepare them in a
```
Comment (stevhliu) [inline docs/source/task_guides/image_classification_lora.mdx:310]:
```suggestion
A collation function is used by [`~transformers.Trainer`] to gather a batch of training and evaluation examples and prepare them in a
```

File: docs/source/task_guides/image_classification_lora.mdx [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+Once the fine-tuning is done, share the LoRA parameters with the community like so:
+
+```python
+repo_name = f"sayakpaul/{model_name}-finetuned-lora-food101"
+lora_model.push_to_hub(repo_name)
+```
+
+When calling `push_to_hub()` on the `lora_model`, only the LoRA parameters along with any modules specified in `modules_to_save`
+are saved. Take a look at the [trained LoRA parameters](https://huggingface.co/sayakpaul/vit-base-patch16-224-in21k-finetuned-lora-food101/blob/main/adapter_model.bin).
+You'll see that it's only 2.6 MB! This greatly helps with portability especially when using a very large model to fine-tune (such as [BLOOM](https://huggingface.co/bigscience/bloom).
```
Comment (stevhliu) [inline docs/source/task_guides/image_classification_lora.mdx:366]:
```suggestion
You'll see that it's only 2.6 MB! This greatly helps with portability, especially when using a very large model to fine-tune (such as [BLOOM](https://huggingface.co/bigscience/bloom)).
```

File: docs/source/task_guides/image_classification_lora.mdx [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+## Share your model and run inference
+
+Once the fine-tuning is done, share the LoRA parameters with the community like so:
+
+```python
+repo_name = f"sayakpaul/{model_name}-finetuned-lora-food101"
+lora_model.push_to_hub(repo_name)
+```
+
+When calling `push_to_hub()` on the `lora_model`, only the LoRA parameters along with any modules specified in `modules_to_save`
```
Comment (stevhliu) [inline docs/source/task_guides/image_classification_lora.mdx:364]:
```suggestion
When calling [`~transformers.PreTrainedModel.push_to_hub`] on the `lora_model`, only the LoRA parameters along with any modules specified in `modules_to_save`
```

File: docs/source/task_guides/image_classification_lora.mdx [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+are saved. Take a look at the [trained LoRA parameters](https://huggingface.co/sayakpaul/vit-base-patch16-224-in21k-finetuned-lora-food101/blob/main/adapter_model.bin).
+You'll see that it's only 2.6 MB! This greatly helps with portability especially when using a very large model to fine-tune (such as [BLOOM](https://huggingface.co/bigscience/bloom).
+
+Next, let's see how to load the LoRA updated parameters along with our base model for inference. When you wrap a base model
+with `PeftModel` that modifications are DONE in place. So to mitigate any concerns that might stem from in place modifications,
```
Comment (stevhliu) [inline docs/source/task_guides/image_classification_lora.mdx:369]:
```suggestion
with `PeftModel`, modifications are done *in-place*. To mitigate any concerns that might stem from in-place modifications,
```

File: docs/source/task_guides/image_classification_lora.mdx [hunk 6]
```diff
@@ -0,0 +1,431 @@
...
+initialize the base model just like you did earlier and construct the inference model.
+
+```python
+from peft import PeftConfig, PeftModel
+
+
+config = PeftConfig.from_pretrained(repo_name)
+model = model = AutoModelForImageClassification.from_pretrained(
```
Comment (stevhliu) [inline docs/source/task_guides/image_classification_lora.mdx:377]:
```suggestion
model = AutoModelForImageClassification.from_pretrained(
```

File: docs/source/task_guides/image_classification_lora.mdx [hunk 7]
```diff
@@ -0,0 +1,431 @@
...
+
+LoRA achieves this reduction by adding low-rank "update matrices" to specific blocks of the model, such as the attention
+blocks. During fine-tuning, only these matrices are trained, while the original model parameters are left unchanged.
+At inference time, the update matrices are merged with the original model parameters to produce the final classification result.
+
+For more information on LoRA, please refer to the [original LoRA paper](https://arxiv.org/abs/2106.09685).
+
+## Install dependencies
+
+Install the libraries required for model training. To ensure you have access to all the latest features of 🤗 PEFT,
```
[the remark this answers is not in the document]
Comment (MKhalusova) [reply docs/source/task_guides/image_classification_lora.mdx]: Actually, we no longer need to install from source to run this example, so I changed to installing a stable version.

File: docs/source/task_guides/image_classification_lora.mdx [hunk 8]
```diff
@@ -0,0 +1,431 @@
...
+`r`: The dimension used by the LoRA update matrices.
+`alpha`: Scaling factor.
+`bias`: Specifies if the `bias` parameters should be trained. `None` denotes none of the `bias` parameters will be trained.
+
+`r` and `alpha` together control the total number of final trainable parameters when using LoRA, giving you the flexibility
+to balance a trade-off between end performance and compute efficiency.
+
+By looking at the number of trainable parameters, you can see how many parameters we're actually training. Since the goal is
+to achieve parameter-efficient fine-tuning, you should expect to see fewer trainable parameters in the `lora_model`
+in comparison to the original model, which is indeed the case here.
```
[the remark this answers is not in the document]
Comment (MKhalusova) [reply docs/source/task_guides/image_classification_lora.mdx:250]: If you mean printing out actual parameters, the table will be rather big, and I don't know if it adds that much value. If you mean the percentage values, then the number 0.77% is repeated a couple of times in the guide. I feel it's enough.

**Review by pacman100 — APPROVED** [review]
Thank you @MKhalusova for adding the image classification example! ✨

Based on issues and usage, I believe the next examples should be around NLP tasks related to CLM/Seq2Seq.

**Comment by MKhalusova** [comment]
@pacman100 good idea! eventually, we'll move all the examples, this is just the first, but it's good to know how to prioritize them :)
